sdk/ml/azure-ai-ml/azure/ai/ml/_schema/schedule/create_job.py

```diff
@@ -32,6 +37,10 @@ def _serialize(self, value, attr, obj, **kwargs):
 
         This function is overwrite because we need job can be dumped inside schedule.
         """
+        from azure.ai.ml.entities._builders import BaseNode
+        if isinstance(value, BaseNode):
+            # Dump as Job to avoid missing field.
+            value = value._to_job()
```

Comment:
hhh so this is the hack?

Reply:
yes, the magic 2 lines, and cannot be put anywhere else =D....